docs/howtos/optax_update_guide.rst

```diff
@@ -0,0 +1,263 @@
+Optax Update Guide
+==================
+
+Code samples below are executable in
+`Colab <https://colab.research.google.com/github/google/flax/blob/main/docs/notebooks/optax_update_guide.ipynb>`_.
```

Comment:
What's the user journey where someone wants to run this Colab? (as opposed to copying code from the RTD page). I ask because I'd rather avoid having to remember to update two places if we change a variable name in this doc, etc.

Reply:
Or, another version of the question: Should we have colabs for *all* of our HOWTOs? If so, maybe we should find a way to automatically convert them. Or is this one special? I'm hesitant to introduce this new expectation by adding just one guide with a Colab, unless we want to go in that direction more broadly.

Reply:
(But no need to block merging this PR on this discussion, just wanted to bring it up. We can resolve separately)

Reply:
I think the user journey is to provide users with a working example that they can use to experiment with different gradient transformations to replicate their original setup. Of course they could do this on their own starting with the code blocks from the transition guide, but there's still some more code involved and it's annoying having to copy this together and a nicer experience to start with a working Colab.

I first tried to have a Colab that is automatically synced, but then realized two problems:

1. I think it's very useful to have the side-by-side in the update guide. This would require us to create a new Sphinx plugin that knows how to take two cells in the Colab, do some pre-processing, and then put them into a diff html table.
2. The Colab will necessarily contain some different code. For example, I want to provide a more realistic example that can be used int he Colab to play around with optimizer settings in a meaningful way. In RST on the other hand we want to have a cheap setup that does not load anything from `tdfs`. This would probably be a simpler Sphinx plugin replacing some code block with another code block when transcribing the Colab into RST.

If we have a Sphinx plugin implementing the 2 features mentioned above then I'll gladly use it. But it's too much work to implement, and I'd rather have to sync the two documents for the added functionality. I don't expect us to update this code frequently, so it's more about remembering (and even if we forgot to update a small thing, it wouldn't be too bad).

Reply:
Ack. Leaving comment open for better visiblity.